.github/workflows/pr.yml

```diff
@@ -0,0 +1,33 @@
+name: PR
+
+on:
+  pull_request:
+    branches: [main]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  pr:
+    name: PR
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v3
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+          registry-url: https://registry.npmjs.org/
+          cache: 'pnpm'
+      - name: Install dependencies
+        run: pnpm install
+      - name: Run lint
+        run: pnpm run lint
+      - name: Run build
+        run: pnpm run build
```

Comment:
Isn't `pnpm run check` enough for this ?

_(Should be faster than a build, but a build is not that long either)_

Reply:
I wish! Only build will run the Zod schemas (runtime), and also svelte-check will still fail as not all type errors are fixed.

Reply:
The alternative I experimented with was renaming it from a json file to something like `data.js` then typing the file directly. SolidJS does a similar thing: https://github.com/solidjs/solid-site/blob/main/src/pages/Resources/Utilities.data.ts

Reply:
What about a small script that run: 
```ts
import { entrySchema } from './src/lib/schema';
import components from './src/routes/components/components.json';
import tools from './src/routes/toold/tools.json';
import templates from './src/routes/templates/templates.json';

Object.entries({components, tools, templates}).forEach(([name, data]) => {
    console.log(`Validating ${name}...`)
    entrySchema.parse(data);
});
```
It will fail on the first file that is incorrect

Reply:
Ooo I like that!